### PR TITLE
added the controller changes for get audit event endpoints

### PR DESCRIPTION
**Pull request by shriramshankar**
This PR includes the changes for the simple get v1/auditEvents request, with filtering on resourceName, resourceType and the isError field. Filtering on the date field will be done in the next PR.

### CODE REVIEW

File: tests/api/v1/auditEvents/post.js [hunk 1]
```diff
@@ -41,6 +41,18 @@ describe('tests/api/v1/auditEvents/post.js >', () => {
     .end(done);
   });
 
+  /*
+   * Swagger converts the object to an array if swagger.yaml expects the
+   * schema type of the request payload to be an array.
+   */
```
Comment (shriramshankar) [inline tests/api/v1/auditEvents/post.js]: @iamigo Added the test that we talked about but you might want to have a look at the comments above.

File: tests/api/v1/auditEvents/get.js [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+const expect = require('chai').expect;
+const tu = require('../../../testUtils');
+const AuditEvent = tu.db.AuditEvent;
+const u = require('./utils');
+const path = '/v1/auditEvents';
+
+describe('tests/api/v1/auditEvents/get.js >', () => {
+  let token;
+  const auditEvent1 = u.createAuditEventObject('Collector', 'Generator',
+    '2017-01-01', 'true');
```
Comment (iamigo) [inline tests/api/v1/auditEvents/get.js]: add times to all of these datetime values

File: tests/api/v1/auditEvents/post.js (hunk 1 above)
Comment (iamigo) [reply tests/api/v1/auditEvents/post.js]: OK

**Comment by shriramshankar** [comment]
@iamigo @harshkothari410 PR updated.

**Comment by shriramshankar** [comment]
@iamigo @harshkothari410 PR updated.

**Pull request merged**